lms/djangoapps/support/views/course_entitlements.py

```diff
@@ -0,0 +1,144 @@
+        try:
+            username_or_email = self.request.data['user']
+            user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
+            course_uuid = request.data['course_uuid']
+            reason = request.data['reason']
+            comments = request.data.get('comments', None)
+            entitlement = self.get_most_recent_entitlement(user=user, course_uuid=course_uuid)
+        except KeyError as err:
+            return HttpResponseBadRequest(u'The field {} is required.'.format(text_type(err)))
+        except (User.DoesNotExist, CourseEntitlement.DoesNotExist):
+            return HttpResponseBadRequest(
+                u'Could not find entitlement to course {course_uuid} for user {username}'.format(
+                    course_uuid=course_uuid, username=username_or_email
+                )
+            )
+        if entitlement:
+            if entitlement.expired_at is None:
+                return HttpResponseBadRequest(
+                    u"Entitlement for user {user} to course {course} is not expired.".format(
+                        user=entitlement.user.username,
+                        course=entitlement.course_uuid
+                    )
+                )
+            if entitlement.enrollment_course_run is None:
+                return HttpResponseBadRequest(
+                    u"Entitlement for user {user} to course {course} has not been spent on a course run.".format(
+                        user=entitlement.user.username,
+                        course=entitlement.course_uuid
+                    )
+                )
+            unenrolled_run = self.unexpire_entitlement(entitlement)
+            CourseEntitlementSupportDetail.objects.create(
+                entitlement=entitlement, reason=reason, comments=comments, unenrolled_run=unenrolled_run,
+                support_user=support_user
+            )
+            return Response(
+                status=status.HTTP_201_CREATED,
+                data=SupportCourseEntitlementSerializer(instance=entitlement).data
+            )
+        else:
+            return HttpResponseBadRequest(
+                u'Could not find an entitlement for user {username} in course {course}'.format(
+                    username=username_or_email,
+                    course=course_uuid
+                )
+            )
+
+    @method_decorator(require_support_permission)
+    def create(self, request, *args, **kwargs):
+        """ Allows support staff to grant a user a new entitlement for a course. """
+        support_user = request.user
+        try:
+            username_or_email = self.request.data['user']
+            user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
+            course_uuid = request.data['course_uuid']
+            reason = request.data['reason']
+            comments = request.data.get('comments', None)
+            mode = request.data.get('mode', 'verified')
```

Comment:
This is concerning.  Why are we setting a default mode here in the code.  The action should fail if a mode is not provided.  This could lead to confusion down the road.

Reply:
Looks like its just a default? But I agree, seems strange.